### ISSUE TITLE
The driver update started causing issues with TLS in MariaDB

### ISSUE THREAD

**Issue by Govindass** [lucko/LuckPerms#2716]
### Description

mysql connections no longer work properly since LuckPerms driver update when using TLS. this also breaks other plugins that use mysql somehow. downgrading LuckPerms fixes everything.
```
Caused by: java.sql.SQLNonTransientConnectionException: (conn=90) SSL hostname verification failed : IPv4 host "178.63.199.211" doesn't correspond to certificate CN "limework.net" and SAN[{DNS:"creative.limework.net"},{DNS:"grafana.limework.net"},{DNS:"imap.limework.net"},{DNS:"limework.net"},{DNS:"mail.limework.net"},{DNS:"smtp.limework.net"},{DNS:"www.limework.net"}]
This verification can be disabled using the option "disableSslHostnameVerification" but won't prevent man-in-the-middle attacks anymore
at me.lucko.luckperms.lib.mariadb.internal.util.exceptions.ExceptionFactory.createException(ExceptionFactory.java:73) ~[?:?]
at me.lucko.luckperms.lib.mariadb.internal.util.exceptions.ExceptionFactory.create(ExceptionFactory.java:179) ~[?:?]
at me.lucko.luckperms.lib.mariadb.internal.protocol.AbstractConnectProtocol.sslWrapper(AbstractConnectProtocol.java:659) ~[?:?]
at me.lucko.luckperms.lib.mariadb.internal.protocol.AbstractConnectProtocol.createConnection(AbstractConnectProtocol.java:532) ~[?:?]
at me.lucko.luckperms.lib.mariadb.internal.protocol.AbstractConnectProtocol.connectWithoutProxy(AbstractConnectProtocol.java:1380) ~[?:?]
... 16 more
```

### Reproduction steps

Simply use LuckPerms 5.2.31 and MariaDB with TLS.

### Expected behaviour

to just work as before

### Environment details

* Server type/version: All, many of my servers have this issue. from 1.12.2 to 1.16.3, from Java 11 to Java 14
* LuckPerms version: `5.2.31`
* MariaDB 10.4.15 (using MariaDB in luckperms config too)
* I am using Let's Encrypt certificate, I followed this tutorial for the setup: https://kitson-consulting.co.uk/blog/configure-mysql-mariadb-ssl-tls

**Comment by lucko** [comment]
It seems that the IPv4 host isn't included on the cert?

Perhaps try using the hostname in the config instead?

**Comment by Govindass** [comment]
seems to be fixed in 5.2.36
